rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerRestClient.java

```diff
@@ -74,17 +84,31 @@
 
     private String restEndpointURL;
 
-    private ClientHttpEngine httpEngine;
-
     private ResteasyProviderFactory providerFactory;
 
+    private static ClientHttpEngine httpEngine;
+
+    private static SSLContext sslContext;
+
     public SchedulerRestClient(String restEndpointURL) {
         this(restEndpointURL, null);
     }
 
     public SchedulerRestClient(String restEndpointURL, ClientHttpEngine httpEngine) {
-        this.restEndpointURL = restEndpointURL;
+
+        if (sslContext == null) {
+            setBlindTrustSSLContext();
+        }
+
```

Comment:
alternatively you can do:
this.httpEngine = httpEngine;

if (this.httpEngine == null) {

this.httpEngine = ...
}

Not a big deal, but modifying parameters should be avoided.

Reply:
I prefer keeping it like that to avoid adding an else block.
Actually, I modified a null parameter.

Reply:
ok